crates/red_knot/tests/utils.rs

```diff
@@ -0,0 +1,42 @@
+// Allow dead code because the helpers are compiled for each test using them and
+// not all of them use all code in here.
+#![allow(dead_code)]
+
+/// Sets an environment variable for the duration this struct is in scope.
+pub(crate) struct EnvVarGuard {
```

Comment:
This is actually not sufficient because cargo runs tests concurrently. So we also need some form of a lock to avoid that multiple tests hold on to the same env variable. Ugh, this is getting out of hand.